SnapshotGenerator/src/RandomExtensions.cs

```diff
@@ -0,0 +1,115 @@
+// from https://github.com/martindevans/CasualGodComplex
+
+using System;
+using System.Collections.Generic;
+
+namespace Demo
+{
+    static class RandomExtensions
+    {
+        /// <summary>
+        /// Generates a sequence of values from a normal distribution, using Box-Muller
+        /// https://en.wikipedia.org/wiki/Box%E2%80%93Muller_transform
+        /// </summary>
+        /// <param name="random">A (uniform) random number generator</param>
+        /// <param name="standardDeviation">The standard deviation of the distribution</param>
+        /// <param name="mean">The mean of the distribution</param>
+        /// <returns>A normally distributed value</returns>
+        public static IEnumerable<float> NormallyDistributedSingles(this Random random, float standardDeviation, float mean)
+        {
+            while(true)
+            {
+                var u1 = random.NextDouble(); //these are uniform(0,1) random doubles
+                var u2 = random.NextDouble();
+                
+                var x1 = Math.Sqrt(-2.0 * Math.Log(u1));
+                var x2 = 2.0 * Math.PI * u2;
+                var z1 = x1 * Math.Sin(x2); //random normal(0,1)
+                yield return (float)(z1 * standardDeviation + mean);
```

Comment:
It looks odd to do two separate but related returns here - it may confuse you in the future (it confuses me now :p), why not return a struct?

Reply:
Seems dryer to use yield return than creating a class, it's specified here https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/yield

Reply:
I agree that using yield return is good, but, to help the argument - what is the difference between the first and the second yield returns here?